lib/cocina/models/collection_identification.rb

```diff
@@ -7,7 +7,7 @@ class CollectionIdentification < Struct
       # Unique identifier in some other system. This is because a large proportion of what is deposited in SDR, historically and currently, are representations of objects that are also represented in other systems. For example, digitized paper and A/V collections have physical manifestations, and those physical objects are managed in systems that have their own identifiers. Similarly, books have barcodes, archival materials have collection numbers and physical locations, etc. The sourceId allows determining if an item has been deposited before and where to look for the original item if you're looking at its SDR representation. The format is: "namespace:identifier"
 
       # example: sul:PC0170_s3_Fiesta_Bowl_2012-01-02_210609_2026
-      attribute? :sourceId, Types::Strict::String
+      attribute? :sourceId, SourceId.optional
```

Comment:
can you explain the "optional" here?  what does it mean?  How was this expressed in `openapi.yml` to get here?  Why wasn't it "optional" before this change?

Reply:
Ah, thanks for asking! I added this late yesterday after discovering a Cocina validation explosion on sdr-infra when testing this branch against stage. Here's what I saw there:

```shell
Validating Dro [▣▣▣▣▣▣▣▣▣▣▣▣▣▣▣▣▣▣▣⬚⬚⬚⬚⬚⬚⬚⬚⬚⬚⬚⬚] (61% (26000/42112), rate: 85.24/s, mean rate: 352.52/s,  1m13s total, ETA: 16:42:32)
/usr/local/rvm/gems/ruby-3.1.2/gems/dry-types-1.7.0/lib/dry/types/schema.rb:332:in `rescue in block in resolve_unsafe': Dry::Struct::
Error: [Cocina::Models::Identification.new] nil (NilClass) has invalid type for :barcode violates constraints (type?(String, nil) AND
 format?(/^36105[0-9]{9}$/, nil) failed) (Parallel::UndumpableException)
```

At the time, I interpreted this as, "I need to add code to allow barcodes to not only be missing but also to be nil." This is what the `.optional` method in dry-types does.

**But** maybe the problem here wasn't in the code and is instead anomalous stage data and I should try undoing this particular change and re-running against a bunch of objects in prod? (This change is a tiny part of the PR, so will be easy to remove.) If you don't have a hunch about this, who should we tag in to advise?

Reply:
I can no longer reproduce this so I yanked the change around generating optional custom types.